src/kcas.ml

```diff
@@ -294,12 +294,17 @@ let update_as g loc f casn state' l r =
   [@@inline]
 
 let update_as g loc f (casn, cass) =
+  let x = loc.id in
   match cass with
   | NIL -> update_as0 g loc f casn NIL NIL
+  | CASN (a, _, NIL, _) as cass when x < a.id ->
+      update_as0 g loc f casn NIL cass
+  | CASN (a, _, _, NIL) as cass when a.id < x ->
+      update_as0 g loc f casn cass NIL
```

Comment:
These implement the same presorted access optimization as is done with the `Op` API.